contracts/receivables/RealWorldReceivable.sol

```diff
@@ -0,0 +1,189 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC721/ERC721Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC721/extensions/ERC721EnumerableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC721/extensions/ERC721URIStorageUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC721/extensions/ERC721BurnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/CountersUpgradeable.sol";
+import "../Errors.sol";
+import "./RealWorldReceivableStorage.sol";
+
+/**
+ * @title RealWorldReceivable
+ * @dev ERC721 tokens that represent off-chain payable receivables
+ */
+contract RealWorldReceivable is
+    RealWorldReceivableStorage,
+    Initializable,
+    ERC721Upgradeable,
+    ERC721EnumerableUpgradeable,
+    ERC721URIStorageUpgradeable,
+    ERC721BurnableUpgradeable,
+    AccessControlUpgradeable
+{
+    using CountersUpgradeable for CountersUpgradeable.Counter;
+
+    // The status of the receivable.
+    enum Status {
+        Unpaid,
+        Paid,
+        PartiallyPaid
+    }
```

Comment:
We probably want to add a status "Voided" to be used to allow the owner to mark it void when there is a mistake in the uploaded record.

Reply:
Do we want to just encourage them to burn the Receivable instead?

Reply:
Sounds good.